src/boot/efi/boot.c

```diff
@@ -1607,8 +1607,9 @@ static void config_load_defaults(Config *config, EFI_FILE *root_dir) {
         if (err == EFI_SUCCESS) {
                 /* Unset variable now, after all it's "one shot". */
                 (void) efivar_set(LOADER_GUID, L"LoaderConfigTimeoutOneShot", NULL, EFI_VARIABLE_NON_VOLATILE);
-
-                config->force_menu = true; /* force the menu when this is set */
+                /* If the variable was set to "0" or "menu-hidden", then we should not show the menu. */
+                if (config->timeout_sec > 0)
+                        config->force_menu = true;
```

Comment:
`config->force_menu` is now useless, so just drop it everywhere.

Reply:
I think that this would still be useful if we can force showing the menu when the correct key is pressed (space, I think). There's a part of code that seems to evaluate that https://github.com/systemd/systemd/blob/4cff70b5bbbe197cc98b5d083a6a03339210d6bb/src/boot/efi/boot.c#L2699-L2701
However, I do not see where the key press is evaluated as `force_menu` is only present in
```
./src/boot/efi/boot.c:93:        bool force_menu;
./src/boot/efi/boot.c:1612:                        config->force_menu = true;
./src/boot/efi/boot.c:2700:        if (config.force_menu || config.timeout_sec > 0)
```
As for:
> I presume to provide a way to force the menu to be visible (indefinitely) as the defaults are a timeout of 0 (without a menu)

Here we are setting a one-shot value, I doubt that it would be much of a problem to now show the menu when the user previously set the one-shot timeout to zero.

Let's see what @poettering think about this.

Reply:
That bool is only set when `LoaderConfigTimeoutOneShot` is found. If you have a timeout, you will see the menu regardless. And for the 0s oneshot timeout you're changing things to now skip the menu, so the var has become useless.

Reply:
Yes, I know that. As I said, it would be still useful *if we can force showing the menu when the correct key is pressed*, that's my only concern before dropping it. Or are you saying that the menu would be still shown if the space key is pressed regardless of `LoaderConfigTimeoutOneShot` being set?

Reply:
Ah yes, I realized now, so it makes sense to just drop `force_menu` everywhere.